File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        
        int n = s.length();
        
        int result = 0;
        
        List<Character> visited =  new ArrayList<Character>();
        
        for(int i=0; i<n; i++) {
            for(int j=i; j<n; j++) {
                if(visited.contains(s.charAt(j))) {
                    visited.clear();
                    break;
                }
                else {
                    visited.add(s.charAt(j));
                    if (result < j - i + 1) {
                        result = j - i + 1;
                    }
                }
            }
        }
        return result;
    }
}

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        
        # faster solution
        left = 0
        right = 0
        result = 0
        mySet = set()
        
        while right < len(s):
            while s[right] in mySet:
                mySet.remove(s[left])
                left += 1
                
            mySet.add(s[right])
            result = max(result, right - left + 1)
            
            right += 1
            
        return result
        
        
        
#         slow solution
#         temp = ""
#         result = 0    
#         left = 0
    
#         for left in range(len(s)):
#             temp = ""
#             right = left 
#             currentSum = 0
            
#             while right < len(s):
#                 if s[right] in temp:
#                     break
#                 else:
#                     temp += s[right]
#                     currentSum += 1
#                 right += 1
#             result = max(result, currentSum)
        
#         return result